### DISCUSSION TITLE
WSL 脚本运行问题：命令行选项无效 -SetDefaultUser

### DISCUSSION THREAD

**Discussion by yuanyi-thu** [physics-data/faq#26] [Q&A] [answered]
其中第60行
[WSL-SetDefaultUser physics-data-wsl debian
](url)
运行结果是“命令行选项无效 -SetDefaultUser”

**Comment by jiegec** [comment, marked as answer]
可以贴一下屏幕截图吗？

**Reply by yuanyi-thu** [reply]
<img width="454" alt="1" src="https://user-images.githubusercontent.com/68137215/129465132-865477a4-cf8a-407d-80ec-2d6de5ba431a.png">

**Reply by jiegec** [reply]
WSL和-SetDefaultUser之间没有空格，并且这个函数是安装脚本里自定义的：

```
Function WSL-SetDefaultUser ($distro, $user) { Get-ItemProperty Registry::HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Lxss\*\ DistributionName | Where-Object -Property DistributionName -eq $distro | Set-ItemProperty -Name DefaultUid -Value ((wsl -d $distro -u $user -e id -u) | Out-String); };
```

**Reply by yuanyi-thu** [reply]
好的谢谢！

**Comment by Berrysoft** [comment]
所以为啥不直接运行这个脚本呢

**Reply by yuanyi-thu** [reply]
因为之前直接运行多次出现了错误，所以进去逐行操作了一下，现在没问题了
之前大概是这么个情况
<img width="471" alt="4" src="https://user-images.githubusercontent.com/68137215/129466104-52d828ec-0be5-404c-9e2f-933691d61477.png">

**Reply by Berrysoft** [reply]
你的电脑有wsl2但是没开虚拟化，因此脚本会默认使用wsl2导入，导致导入失败。
